src/components/button/button.js

```diff
@@ -162,20 +162,15 @@ function MdButtonDirective($mdButtonInkRipple, $mdTheming, $mdAria, $timeout) {
     });
 
     if (!element.hasClass('md-no-focus')) {
-      // restrict focus styles to the keyboard
-      scope.mouseActive = false;
-      element.on('mousedown', function() {
-        scope.mouseActive = true;
-        $timeout(function(){
-          scope.mouseActive = false;
-        }, 100);
-      })
-      .on('focus', function() {
-        if (scope.mouseActive === false) {
+
+      element.on('focus', function() {
+        // Only show the focus effect when being focused through keyboard interaction.
+        if ($mdInteraction.getLastInteractionType() === 'keyboard') {
```

Comment:
This does not change anything to the existing behavior, just that we now check the global interaction to detect the last interaction.

Focusing manually per `.focus` does not work right now as well, because the `mousedown` event didn't get fired.

This is the approach we wanted to go with the `$mdInteraction` service, as proposed by Marcy. Similar to #9827